core/include/drivers/rtc.h

```diff
@@ -161,8 +161,8 @@ static inline TEE_Result rtc_set_time(struct optee_rtc_time *tm)
 	if (!rtc_device || !rtc_device->ops->set_time)
 		return TEE_ERROR_NOT_SUPPORTED;
 
-	if ((tm->tm_year < rtc_device->range_min.tm_year ||
-	     tm->tm_year > rtc_device->range_max.tm_year) || tm->tm_mon >= 12 ||
+	if (tm->tm_year < rtc_device->range_min.tm_year ||
+	    tm->tm_year > rtc_device->range_max.tm_year || tm->tm_mon >= 12 ||
```

Comment:
To make things simple, can use  `rtc_timecmp()` to check against device range full time bounds:
```c
	if (rtc_timecmp(tm, &rtc_device->range_min) < 0 ||
	    rtc_timecmp(tm, &rtc_device->range_max) > 0 || tm->tm_mon >= 12 ||